docs/content/preview/releases/techadvisories/ta-23476.md

```diff
@@ -0,0 +1,169 @@
+---
+title: TA-23476
+headerTitle: YCQL currenttimestamp() precision
+headcontent: 01 Jan 2024
+type: docs
+showRightNav: true
+cascade:
+  unversioned: true
+menu:
+  preview_releases:
+    identifier: ta-23476
+    weight: 1
+rightNav:
+  hideH2: true
+type: docs
+---
+
+|          Product           |  Affected Versions  |  Related Issues   | Fixed In |
+| :------------------------- | :------------------ | :---------------- | :------- |
+| {{<product "ycql">}}       | {{<release "All">}} | {{<issue 11052>}}, {{<issue 23476>}} | TBD |
+
+## Description
+
+In YCQL, timestamp data inserted with the `currenttimestamp()` function may not be returned for queries with an equality condition on the timestamp. If the timestamp is part of the primary key, this also affects index scans as the table lookup for an index scan internally uses an equality condition to lookup the corresponding table rows. The data can still be retrieved as expected by primary key or index-only scan queries with no condition or inequality conditions on the timestamp column.
+
+The root cause is that in YCQL, the timestamp type has millisecond precision, to align with vanilla Cassandra, ensuring compatibility with Cassandra drivers and tools. However, YCQL can internally store data with microsecond precision, adjusting the precision to milliseconds when data is sent or received through the YCQL API. The discrepancy between the wire protocol's millisecond precision and the storage system's microsecond precision can lead to unexpected behavior.
+
+## Mitigation
+
+There are two key aspects to mitigating the issues caused by the discrepancy between the YCQL wire protocol's millisecond precision and the storage system's microsecond precision:
+
+1. Ensure new writes use the correct precision:
+
+    1. Use totimestamp(now()) instead of currenttimestamp() to generate millisecond-precision timestamp values that are compatible with the wire protocol.
```

Comment:
```suggestion
    Use totimestamp(now()) instead of currenttimestamp() to generate millisecond-precision timestamp values that are compatible with the wire protocol.
```